src/reactviews/pages/SchemaCompare/SchemaCompareStateProvider.tsx

```diff
@@ -0,0 +1,149 @@
+                        taskExecutionMode: taskExecutionMode,
+                        deploymentOptions: deploymentOptions,
+                    });
+                },
+                schemaCompareGenerateScript: function (
+                    operationId: string,
+                    targetServerName: string,
+                    targetDatabaseName: string,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaCompareGenerateScript",
+                        {
+                            operationId: operationId,
+                            targetServerName: targetServerName,
+                            targetDatabaseName: targetDatabaseName,
+                            taskExecutionMode: taskExecutionMode,
+                        },
+                    );
+                },
+                schemaComparePublishDatabaseChanges: function (
+                    operationId: string,
+                    targetServerName: string,
+                    targetDatabaseName: string,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaComparePublishDatabaseChanges",
+                        {
+                            operationId: operationId,
+                            targetServerName: targetServerName,
+                            targetDatabaseName: targetDatabaseName,
+                            taskExecutionMode: taskExecutionMode,
+                        },
+                    );
+                },
+                schemaComparePublishProjectChanges: function (
+                    operationId: string,
+                    targetProjectPath: string,
+                    targetFolderStructure: mssql.ExtractTarget,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaComparePublishProjectChanges",
+                        {
+                            operationId: operationId,
+                            targetProjectPath: targetProjectPath,
+                            targetFolderStructure: targetFolderStructure,
+                            taskExecutionMode: taskExecutionMode,
+                        },
+                    );
+                },
+                schemaCompareIncludeExcludeNode: function (
+                    operationId: string,
+                    diffEntry: mssql.DiffEntry,
+                    includeRequest: boolean,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaCompareIncludeExcludeNode",
+                        {
+                            operationId: operationId,
```

Comment:
I wonder if operationId is something that the React layer actually needs to know, or if the controller can hold onto that since each schema compare window will have its own controller instance.

Reply:
I think the `operationId` can stay out of the react layer and stay with the controller too.